### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by He-Ro** [moosotc/llpp#54]
I get the following error when executing build.sh:
```
./link.c: In function ‘ml_init’:
./link.c:4472:14: error: pointer targets in assignment differ in signedness [-Wpointer-sign]
         data = pdf_lookup_substitute_font (state.ctx, 0, 0, 0, 0, &len);
              ^
```
I see that the variable `$ccopt` has `-Wno-pointer-sign` set, but [Line 54](https://github.com/moosotc/llpp/blob/master/build.sh#L54) does not use `$ccopt`.
Adding `$ccopt` to the line makes llpp compile for me.
So was this intentional, or is this a bug? If it is a bug, then line 51 should probably also be fixed.

**Comment by moosotc** [comment]
He-Ro <notifications@github.com> writes:

> I get the following error when executing build.sh:
> ```
> ./link.c: In function ‘ml_init’:
> ./link.c:4472:14: error: pointer targets in assignment differ in signedness [-Wpointer-sign]
>          data = pdf_lookup_substitute_font (state.ctx, 0, 0, 0, 0, &len);
>               ^
> ```

What commit are you on? I'm interested in sha1s for both mupdf and llpp.

> I see that the variable `$ccopt` has `-Wno-pointer-sign` set, but [Line 54](https://github.com/moosotc/llpp/blob/master/build.sh#L54) does not use `$ccopt`.
> Adding `$ccopt` to the line makes llpp compile for me.

$ccopt is only used for lablGL code, i.e. something I'm not terribly
comfortable of mindlessly "fixing". When link.c is compiled a different
(more strict) set of options is used. What I'm trying to say here is
link.c should be fixed if there's an issue.

> So was this intentional, or is this a bug? If it is a bug, then line 51 should probably also be fixed.

It's intentional.

-- 
mailto:moosotc@gmail.com

**Comment by He-Ro** [comment]
Wow, that was a stupid error. I was using an old checkout of llpp to build it again. But I only pulled the new changes from llpp, but not from mupdf. So it was building against an old checkout of mupdf. I also tried it with the version shipped with Arch Linux, which is currently version 1.11 or 34e18d127a02146e3415b33c4b67389ce1ddb614, but that is also not compatible.
Using the commit of mupdf mentioned in the build instructions lead to a successful build.
